File: functions/javascript/src/constants/quota.ts
/**
 * quota.ts — 生成回数のクォータ定数
 *
 * remaining_sentences / remaining_quizzes の初期値・リセット値を定義。
 * dailyBatch, verifySubscription, handlePlayNotification, handleAppStoreNotification で使用。
 */

/** free ユーザーの日次リセット値（JST 0:00） */
export const FREE_DAILY_SENTENCES = 5;

/**
 * クイズの日次上限は 2026-08-25 に撤廃した（generateQuiz は remaining_quizzes を
 * 読まないし減らさない）。出題対象は SRS で期日を迎えた自分の例文だけなので、
 * 実質的な上限は例文側のクォータが決めている。
 * remaining_quizzes フィールド自体は既存ドキュメントの形を保つために書き続けており、
 * 以下の値はそのリセット用。読み手が居なくなったらフィールドごと削除してよい。
 */
export const FREE_DAILY_QUIZZES = 5;

/**
 * premium ユーザーの日次リセット値（JST 0:00）
 *
 * 例文は 2026-08-09 に 5 → 10、2026-08-25 に 10 → 20 へ引き上げ。後者は
 * 「1日1文」から「たくさん触れる」へブランドを寄せる方針に合わせたもので、
 * free 5 に対して 4 倍の差を作る。月額600円・Apple手取り510円に対し、
 * 20文/日をフル消化しても AI コストは月約275円（例文 $0.00246/回 実測）で粗利は残る。
 * クイズは上限そのものを撤廃した（FREE_DAILY_QUIZZES のコメント参照）。
 */
export const PREMIUM_DAILY_SENTENCES = 20;
/** 上限としては機能しない。FREE_DAILY_QUIZZES のコメント参照。 */
export const PREMIUM_DAILY_QUIZZES = 5;

/**
 * 新規ユーザーへのプレミアム体験トライアル期間（日）。
 * 2日にして「連続して使う」体験を作る。期限は premium_trial_expires_at。
 * 期間中は機能・回数とも課金 premium と完全に同じ（utils/premium.ts）。
 */
export const PREMIUM_TRIAL_DAYS = 2;

/**
 * premium_trial_remaining の付与値（凍結した互換値）。
 *
 * トライアルは期間制に一本化済みで、サーバはこの値を読まないし減らさない。
 * ただし 1.3.14（現行ストア版）までのクライアントは「残回数 <= 1 なら体験最終回」と見なして
 * 設定テーマを「おまかせ」へ戻すため、書かないと体験中の1回目でテーマが消える。
 * 該当バージョンが行き渡らなくなったらフィールドごと削除してよい。
 */
export const PREMIUM_TRIAL_SENTENCES = PREMIUM_DAILY_SENTENCES * PREMIUM_TRIAL_DAYS;

File: functions/go/internal/quota/quota.go
// Package quota は生成回数のクォータ定数。
// functions/javascript/src/constants/quota.ts の移植。**両者を必ず一致させること。**
package quota

const (
	// FreeDailySentences は free ユーザーの日次リセット値（JST 0:00）。
	FreeDailySentences = 5

	// FreeDailyQuizzes はクイズの日次リセット値。
	// クイズの日次上限は 2026-08-25 に撤廃済みで、この値は既存ドキュメントの
	// 形を保つためだけに書き続けている（読み手は居ない）。
	FreeDailyQuizzes = 5

	// PremiumDailySentences は premium ユーザーの日次リセット値。
	//
	// 2026-09-12 に 20 から「無制限」へ切り替えた。premium の例文は静的コーパスから
	// 出すようになり（produce.go の Corpus 経路）、1本あたりの限界コストがほぼ 0 に
	// なったため、回数で絞る理由が無くなった。
	//
	// premium は消費もしない（generateThaiSentence / deliverDailySentence は
	// remaining_sentences に触らない）。この値は「無制限」を示す表示用の残数として
	// 毎日入れ直すだけで、上限としては機能しない。free だけが実際に消費する。
	PremiumDailySentences = 9999

	// PremiumDailyQuizzes は上限としては機能しない。FreeDailyQuizzes 参照。
	PremiumDailyQuizzes = 5

	// PremiumTrialDays は新規ユーザーへのプレミアム体験トライアル期間（日）。
	PremiumTrialDays = 2

	// PremiumTrialSentences は premium_trial_remaining の付与値（凍結した互換値）。
	// サーバは読まないが、1.3.14 までのクライアントがテーマ判定に使う。
	// 無制限化する前の PremiumDailySentences(20) * PremiumTrialDays(2) の値をそのまま置く。
	PremiumTrialSentences = 40
)
